types/assert/index.d.ts

```diff
@@ -5,63 +5,103 @@
+    function equal(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use notStrictEqual() instead. */
+    function notEqual(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use deepStrictEqual() instead. */
+    function deepEqual(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use notDeepStrictEqual() instead. */
+    function notDeepEqual(actual: any, expected: any, message?: string | Error): void;
+    function strictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+    function notStrictEqual(actual: any, expected: any, message?: string | Error): void;
+    function deepStrictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+    function notDeepStrictEqual(actual: any, expected: any, message?: string | Error): void;
+
+    function throws(block: () => any, message?: string | Error): void;
+    function throws(block: () => any, error: AssertPredicate, message?: string | Error): void;
+    function doesNotThrow(block: () => any, message?: string | Error): void;
+    function doesNotThrow(block: () => any, error: RegExp | Function, message?: string | Error): void;
+
+    function ifError(value: any): asserts value is null | undefined;
+
+    function rejects(block: (() => Promise<any>) | Promise<any>, message?: string | Error): Promise<void>;
+    function rejects(
+        block: (() => Promise<any>) | Promise<any>,
+        error: AssertPredicate,
+        message?: string | Error,
+    ): Promise<void>;
+    function doesNotReject(block: (() => Promise<any>) | Promise<any>, message?: string | Error): Promise<void>;
+    function doesNotReject(
+        block: (() => Promise<any>) | Promise<any>,
+        error: RegExp | Function,
+        message?: string | Error,
+    ): Promise<void>;
+
+    function match(value: string, regExp: RegExp, message?: string | Error): void;
+    function doesNotMatch(value: string, regExp: RegExp, message?: string | Error): void;
+
+    const strict: Omit<
+        typeof assert,
+        | 'strict'
+        | 'deepEqual'
+        | 'notDeepEqual'
+        | 'equal'
+        | 'notEqual'
+        | 'ok'
+        | 'strictEqual'
+        | 'deepStrictEqual'
+        | 'ifError'
+    > & {
+        (value: any, message?: string | Error): asserts value;
+        strict: typeof strict;
+        deepEqual: typeof deepStrictEqual;
+        notDeepEqual: typeof notDeepStrictEqual;
+        equal: typeof strictEqual;
+        notEqual: typeof notStrictEqual;
+        ok(value: any, message?: string | Error): asserts value;
+        strictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+        deepStrictEqual<T>(actual: any, expected: T, message?: string | Error): asserts actual is T;
+        ifError(value: any): asserts value is null | undefined;
```

Comment:
See <https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49469>:

```suggestion

        // Mapped types and assertion functions are incompatible?
        // TS2775: Assertions require every name in the call target
        // to be declared with an explicit type annotation.
        ok: typeof ok;
        strictEqual: typeof strictEqual;
        deepStrictEqual: typeof deepStrictEqual;
        ifError: typeof ifError;
```

Reply:
Done :heavy_check_mark: